content/browser/web_contents/web_contents_impl.cc

```diff
@@ -1096,6 +1096,13 @@ std::unique_ptr<WebContentsImpl> WebContentsImpl::CreateWithOpener(
     opener = opener_rfh->frame_tree_node();
   std::unique_ptr<WebContentsImpl> new_contents(
       new WebContentsImpl(params.browser_context));
+
+  // Forward flag indicating that this web-contents is being created
+  // for a replay.io recording.
+  if (params.record_replay_for_recording) {
+    new_contents->record_replay_for_recording_ = true;
```

Comment:
Nit: Why not call it `recording` instead of `for_recording` (it took me a while to understand what this means)?

Reply:
I'm not married to the name, but it's a pain to change it all now.  Gonna go with this for now and we can bikeshed it later.